apps/web/components/ActionButtons.tsx

```diff
@@ -56,6 +61,20 @@ export function ActionButtons(props: {
     setIsTrashing(false);
   }, [threadId, refetch]);
 
+  const [isReadUnreading, setisReadUnreading] = useState(false);
+  const [isStateRead, setIsStateRead] = useState(isRead);
```

Comment:
Don't think you want to use this local state at all. You want to know the state of the email from the row itself. We use it other places to mark the row color as read.